### PR TITLE
feat(mint-time): add mint-time show under creator

### LINKED ISSUE #2219: Add time when was collection minted
Probably add it somewhere under creator (pic1) with the option to turn it off in settings. 
Add it as relative time (pic2)

image

image

### PR DESCRIPTION
**Pull request by zhengow**
…reator

**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2219 
- [x] add mint-time show under creator and add choice at setting to control

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [x] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

https://user-images.githubusercontent.com/38389586/155082953-6226aff5-e6ff-4e8c-9f2c-4317f6df0e32.mp4

### CODE REVIEW

File: components/rmrk/Gallery/CollectionItem.vue [hunk 1]
```diff
@@ -257,6 +278,31 @@ export default class CollectionItem extends mixins(ChainMixin, PrefixMixin) {
...
+            account: this.issuer || '',
+          }
+        },
+        fetchPolicy: 'cache-and-network',
+      })
+    } catch (e) {
+      showNotification(`${e}`, notificationTypes.danger)
+      console.warn(e)
+    }
+  }
```
Comment (roiLeo) [inline components/rmrk/Gallery/CollectionItem.vue:304]: Isn't this duplicate from another component?

**Review by roiLeo — COMMENTED** [review]
code looks good, might need refactoring later

File: components/rmrk/Gallery/CollectionItem.vue (hunk 1 above)
Comment (zhengow) [reply components/rmrk/Gallery/CollectionItem.vue:304]: Yes, many codes from "IdentityPopover.vue".
What's the problem? Better to refactor this for both components?

Comment (zhengow) [reply components/rmrk/Gallery/CollectionItem.vue:304]: > 

I can use **_mixin_** to refactor them

Comment (vikiival) [reply components/rmrk/Gallery/CollectionItem.vue:304]: Why such a huge query?

**Review by vikiival — CHANGES_REQUESTED** [review]
Not correct tbh

The issue asked only for when the collection was created

```graphql


query created {
  collectionEntity(id: "900D19DC7D3C444E4C-ROOMS") {
    id
    createdAt
  }
}
```

The task is to extend `collectionById.graphql` with `createdAt` and then format it (this part looks good)

**Comment by zhengow** [comment]
> Not correct tbh
> 
> The issue asked only for when the collection was created
> 
> ```graphql
> query created {
>   collectionEntity(id: "900D19DC7D3C444E4C-ROOMS") {
>     id
>     createdAt
>   }
> }
> ```
> 
> The task is to extend `collectionById.graphql` with `createdAt` and then format it (this part looks good)

got it, thanks

**Comment by yangwao** [comment]
yes please, keep quality code 🥺 and better reuse/refactor components into reusable components. 
we are against duplicating code where we can be smart about it )

**Comment by zhengow** [comment]
> 

Thanks for [vikiival](https://github.com/vikiival), now I'm refactor format function with Vue Mixin, and reuse query to get createdAt data, plz review again.

File: components/rmrk/Gallery/CollectionItem.vue [hunk 2]
```diff
@@ -317,6 +330,7 @@ export default class CollectionItem extends mixins(ChainMixin, PrefixMixin) {
       this.$router.push({ name: 'errorcollection' })
       return
     }
+    this.firstMintDate = collectionEntity.createdAt
```
Comment (vikiival) [inline components/rmrk/Gallery/CollectionItem.vue:333]: where setter 👁️

Comment (zhengow) [reply components/rmrk/Gallery/CollectionItem.vue:333]: Do u mean where is the initialize? It's in the CreatedAtMixin

`export default class CreatedAtMixin extends Vue {
  protected firstMintDate = new Date()

  get formattedTimeToNow() {
    return this.firstMintDate
      ? formatDistanceToNow(new Date(this.firstMintDate), { addSuffix: true })
      : ''
  }
}
`

**Review by vikiival — APPROVED** [review]
Looks legit

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 121.02 USD/KSM ~ 0.826 $KSM
🧗 [EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws](https://kusama.subscan.io/account/EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws)
🔗 [0xa4da27a2b2b50a1e901c6e12bad0e3af9a942f8d96d6b347f9bc1b783bf40971](https://kusama.subscan.io/extrinsic/0xa4da27a2b2b50a1e901c6e12bad0e3af9a942f8d96d6b347f9bc1b783bf40971)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)
